src/components/SearchBar/SearchBarColumn/index.tsx

```diff
@@ -1,16 +1,33 @@
+import {
+    isStringColumnFilterValue,
+    isDateColumnFilterValue,
+    isReferenceColumnFilterValue,
+} from 'src/components/SearchBar/types';
+
 import { DateColumn } from './DateColumn';
+import { ReferenceColumn } from './ReferenceColumn';
 import { StringColumn } from './StringColumn';
-import { SearchBarColumnDateTypeProps, SearchBarColumnProps, SearchBarColumnStringTypeProps } from './types';
+import {
+    SearchBarColumnDateTypeProps,
+    SearchBarColumnProps,
+    SearchBarColumnReferenceTypeProps,
+    SearchBarColumnStringTypeProps,
+} from './types';
 
-export function SearchBarColumn<T>(props: SearchBarColumnProps<T>) {
+export function SearchBarColumn(props: SearchBarColumnProps) {
     const { columnFilterValue } = props;
 
-    switch (columnFilterValue.column.type) {
-        case 'string':
-            return <StringColumn<T> {...(props as SearchBarColumnStringTypeProps)} />;
-        case 'date':
-            return <DateColumn<T> {...(props as SearchBarColumnDateTypeProps)} />;
-        default:
-            return null;
+    if (isStringColumnFilterValue(columnFilterValue)) {
```

Comment:
Rid of type conversion and `as` operator

Reply:
done